### PR TITLE
CLOUDP-125628 security context update

### PR DESCRIPTION
**Pull request by slaskawi**
This Pull Request removes `securityContext` set on container level and moves all the previously used settings to the pod level. Note, that previously the pod level settings didn't make any effect as they were overriden by the container level.

### All Submissions:

* [x] Have you opened an Issue before filing this PR?
* [x] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [x] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

### CODE REVIEW

File: pkg/kube/podtemplatespec/podspec_template.go [hunk 1]
```diff
@@ -160,10 +160,11 @@ func WithSecurityContext(securityContext corev1.PodSecurityContext) Modification
 	}
 }
 
-// DefaultPodSecurityContext returns the default pod security context with FsGroup = 2000
+// DefaultPodSecurityContext returns the default pod security context with uid 2000 and non root
 func DefaultPodSecurityContext() corev1.PodSecurityContext {
-	fsGroup := int64(2000)
```
Comment (slaskawi) [inline pkg/kube/podtemplatespec/podspec_template.go:165]: This is required!!!

File: pkg/kube/podtemplatespec/podspec_template.go [hunk 2]
```diff
@@ -160,10 +160,12 @@ func WithSecurityContext(securityContext corev1.PodSecurityContext) Modification
 	}
 }
 
-// DefaultPodSecurityContext returns the default pod security context with FsGroup = 2000
```
Comment (slaskawi) [inline pkg/kube/podtemplatespec/podspec_template.go:163]: Correct description!

File: controllers/construct/build_statefulset_test.go [hunk 3]
```diff
@@ -87,6 +88,7 @@ func assertStatefulSetIsBuiltCorrectly(t *testing.T, mdb mdbv1.MongoDBCommunity,
 	assert.Equal(t, mongodbDatabaseServiceAccountName, sts.Spec.Template.Spec.ServiceAccountName)
 	assert.Len(t, sts.Spec.Template.Spec.Containers[0].Env, 4)
 	assert.Len(t, sts.Spec.Template.Spec.Containers[1].Env, 1)
+	assert.Equal(t, podtemplatespec.DefaultPodSecurityContext(), *sts.Spec.Template.Spec.SecurityContext)
```
Comment (lsierant) [inline controllers/construct/build_statefulset_test.go:91]: Is it possible that `sts.Spec.Template.Spec.SecurityContext` is nil here? It's good to have require.NotNil before dereferencing if it's a possibility. If it's nil the panic is polluting the logs.

File: Makefile [hunk 4]
```diff
@@ -32,11 +32,8 @@ all: manager
 
 # Run unit tests
 TEST ?= ./pkg/... ./api/... ./cmd/... ./controllers/... ./test/e2e/util/mongotester/...
-ENVTEST_ASSETS_DIR=$(shell pwd)/testbin
 test: generate fmt vet manifests
-	mkdir -p ${ENVTEST_ASSETS_DIR}
```
Comment (mircea-cosbuc) [inline Makefile:37]: Nice catch, I thought this was used by the test actions.

**Review by mircea-cosbuc — APPROVED** [review]
LGTM

File: controllers/construct/build_statefulset_test.go (hunk 3 above)
Comment (slaskawi) [reply controllers/construct/build_statefulset_test.go:91]: In this  unit test, it's not. However, you have a very good point, I should add a test testing a managed security (so that it will be nil). Good spot!

Comment (slaskawi) [reply controllers/construct/build_statefulset_test.go]: Added more tests in this area (and a requested nil/non-nil check).
